packages/flutter_tools/lib/src/asset.dart

```diff
@@ -116,6 +117,32 @@ class AssetBundle {
         manifestDescriptor.containsKey('uses-material-design') &&
         manifestDescriptor['uses-material-design'];
 
+    // Add assets from packages.
+    for (String packageName in packageMap.map.keys) {
+      final Uri package = packageMap.map[packageName];
+      if (package != null && package.scheme == 'file') {
+        String packageManifestPath = package.resolve('../pubspec.yaml').path;
+        Object packageManifest = _loadFlutterManifest(packageManifestPath);
+        if (packageManifest == null)
+          continue;
+        final int result = await _validateFlutterManifest(packageManifest);
+        if (result == 0) {
+          Map<String, dynamic> packageManifestDescriptor = packageManifest;
```

Comment:
You could make this final, if you remove line 134 and then use `packageManifestDescriptor['flutter']` directly in line 138.